cmd/gateway-gcs-anonymous.go

```diff
@@ -111,6 +111,17 @@ func (l *gcsGateway) AnonListObjects(bucket string, prefix string, marker string
 	return fromMinioClientListBucketResult(bucket, result), nil
 }
 
+// AnonListObjectsV2 - List objects in V2 mode, anonymously
+func (l *gcsGateway) AnonListObjectsV2(bucket, prefix, continuationToken string, fetchOwner bool, delimiter string, maxKeys int) (ListObjectsV2Info, error) {
+	// Request V1 List Object to the backend
+	result, err := l.anonClient.ListObjects(bucket, prefix, continuationToken, delimiter, maxKeys)
+	if err != nil {
+		return ListObjectsV2Info{}, s3ToObjectError(traceError(err), bucket)
+	}
+	// translate V1 Result to V2Info
+	return fromMinioClientListBucketResultToV2Info(bucket, result), nil
+}
```

Comment:
> Did you test anon listing on GCS gateway @NitishT ?

Yes @krishnasrinivas GCS backend is tested. Need to test S3 backend. 

> what happens if we call l.anonClient.ListObjectsV2 instead of l.anonClient.ListObjects?

Since `list-type=2` is not supported on GCS, it responds with `Invalid Parameters`